sdk/core/azure-core/inc/azure/core/http/raw_response.hpp

```diff
@@ -76,10 +76,19 @@ namespace Azure { namespace Core { namespace Http {
             response.m_statusCode,
             response.m_reasonPhrase)
     {
+      AZURE_ASSERT(m_bodyStream == nullptr);
       // Copy body
       m_body = response.GetBody();
     }
 
+    /**
+     * @brief Constructs a new `RawResponse`.
+     * @note Transfers ownership of the `RawResponse` to the new `RawResponse`.
+     *
+     * @param response An rvalue reference for moving the raw response.
+     */
+    RawResponse(RawResponse&& response) = default;
```

Comment:
@CaseyCarter does the RawResponse type follow the 0-3-5 rule?

Reply:
No, it doesn't. The class is correct (well, as much as something that's copyable but can't always be copied can be) but seeing that requires inspection and knowledge of the complicated rules for when and how special member functions are defined. I'd add:
```c++
~RawResponse() = default;
RawResponse& operator=(RawResponse const&) = delete;
RawResponse& operator=(RawResponse&&) = delete;
```
here somewhere to make it explicit that we've considered all five special member functions.